src/wikibots/flickr.py

```
@@ -18,44 +26,287 @@ class FlickrBot(BaseBot):
+            return single_photo
+
+        try:
+            start = perf_counter()
+            single_photo = self.flickr_api.get_single_photo_info(photo_id=flickr_photo_id)
+            info(f"Retrieved Flickr photo in {(perf_counter() - start) * 1000:.0f} ms")
+        except (PhotoIsPrivate, ResourceNotFound) as e:
+            warning(f"[{flickr_photo_id}] {e}")
+            self.redis.set(redis_key_photo, 1)
+        except Exception as e:
+            error(f"[{flickr_photo_id}] {e}")
+            time.sleep(60)
```

Ruff (0.8.2)

Undefined name `time`

(F821)